gremlin-go/driver/resultSet.go

```diff
@@ -50,25 +50,60 @@ type channelResultSet struct {
 	statusAttributes map[string]interface{}
 	closed           bool
 	err              error
-	mux              sync.Mutex
+	waitSignal       chan bool
+	channelMux       sync.Mutex
+	waitSignalMux    sync.Mutex
+}
+
+func (channelResultSet *channelResultSet) sendSignal() {
+	// Lock wait
+	channelResultSet.waitSignalMux.Lock()
+	if channelResultSet.waitSignal != nil {
+		channelResultSet.waitSignal <- true
+		channelResultSet.waitSignal = nil
+	}
+	channelResultSet.waitSignalMux.Unlock()
```

Comment:
```suggestion
	channelResultSet.waitSignalMux.Lock()
	defer channelResultSet.waitSignalMux.Unlock()
	if channelResultSet.waitSignal != nil {
		channelResultSet.waitSignal <- true
		channelResultSet.waitSignal = nil
	}
```
Its the same thing basically though so can ignore if not needed